src/main/java/software/sigma/internship/service/StudentServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package software.sigma.internship.service;
+
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+import software.sigma.internship.dao.repo.StudentRepository;
+import software.sigma.internship.user.Student;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+public class StudentServiceImpl implements StudentService {
+
+    private StudentRepository studentRepository;
+
+    @Override
+    public Student save(Student student) {
+        return studentRepository.save(student);
+    }
+
+    @Override
+    public List<Student> fetchList() {
+        return studentRepository.findAll();
+    }
+
+    @Override
+    public Student update(Student student, Long id) {
+        Student studentDB = studentRepository.findById(id).get();
+        if (Objects.nonNull(student.getFirstName())
+                && !"".equalsIgnoreCase((
+                student.getFirstName()))) {
+            studentDB.setFirstName(student.getFirstName());
+        }
+
+        if (Objects.nonNull(student.getLastName())
+                && !"".equalsIgnoreCase((
+                student.getLastName()))) {
+            studentDB.setLastName(student.getLastName());
+        }
+        if (student.getCourse() < 7 && student.getCourse() > 0) {
+            studentDB.setCourse(student.getCourse());
```

Comment:
good point, that we should use validation. But let's move validation to a different class.
Also, it would be good to make the maximum course configurable. Read about @Value and how to read the value from a properties file